File: backend/src/main/java/space/davids_digital/sweetie/orm/entity/VkUserEntity.java
package space.davids_digital.sweetie.orm.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

import java.time.ZonedDateTime;

// TODO original ts file contains sql triggers
@Entity
@Table(name = "vk_users")
public class VkUserEntity {
    @Id
    @Column(name = "id")
    public long id;

    @Column(name = "first_name")
    public String firstName;

    @Column(name = "last_name")
    public String lastName;

    @Column(name = "credits")
    public long credits;

    @Column(name = "last_credit_gain")
    public ZonedDateTime lastCreditGain;

    @Column(name = "usage_plan_id")
    public String usagePlanId;

    @Column(name = "usage_plan_expiry")
    public ZonedDateTime usagePlanExpiry;
}

File: backend/src/main/java/space/davids_digital/sweetie/orm/entity/VkUserEntity.kt
package space.davids_digital.sweetie.orm.entity

import jakarta.persistence.*
import java.time.ZonedDateTime

@Entity
@Table(
    name = "vk_users",
    indexes = [
        Index(
            name = "vk_users__credits_index",
            columnList = "credits"
        )
    ]
)
class VkUserEntity {
    @Id
    @Column(name = "id")
    var id: Long = 0

    @Column(name = "first_name", columnDefinition = "text")
    var firstName: String = ""

    @Column(name = "last_name", columnDefinition = "text")
    var lastName: String = ""

    @Column(name = "credits")
    var credits: Long = 0

    @Column(name = "last_credit_gain")
    var lastCreditGain: ZonedDateTime? = null

    @Column(name = "usage_plan_id", columnDefinition = "text")
    var usagePlanId: String? = null

    @Column(name = "usage_plan_expiry")
    var usagePlanExpiry: ZonedDateTime? = null
}
